packages/linter/src/migrations/update-16-8-0-add-ignored-files/update-16-8-0-add-ignored-files.ts

```diff
@@ -0,0 +1,74 @@
+import { getProjects, Tree } from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+import {
+  findEslintFile,
+  lintConfigHasOverride,
+  updateOverrideInLintConfig,
+} from '../../generators/utils/eslint-file';
+
+// Add `ignoredFiles` pattern to projects using vite, esbuild, and rollup.
+// This is needed because the @nx/dependency-checks lint rule will complain
+// about dependencies used in build config files, even though they are not
+// production dependencies.
+export default function update(tree: Tree) {
+  const projects = getProjects(tree);
+
+  const addIgnorePattern =
+    (ignorePattern: string) => (_options: unknown, projectName: string) => {
+      const project = projects.get(projectName);
+      if (!findEslintFile(tree, project.root)) return;
+      if (
+        lintConfigHasOverride(
+          tree,
+          project.root,
+          (o) =>
+            Array.isArray(o.files)
+              ? o.files.some((f) => f.match(/\.json$/))
+              : !!o.files?.match(/\.json$/),
+          true
+        )
+      ) {
+        updateOverrideInLintConfig(
+          tree,
+          project.root,
+          (o) => !!o.rules?.['@nx/dependency-checks'],
+          (o) => {
+            const value = o.rules['@nx/dependency-checks'];
+            let ruleSeverity: 0 | 1 | 2 | 'error' | 'warn' | 'off';
```

Comment:
This is no longer correct. The current type doesn't support numbers

Reply:
It was giving me type errors if I don't include them. I can cast as any as well.